lib/symboldatabase.cpp

```diff
@@ -4202,8 +4204,41 @@ static const Token* skipPointersAndQualifiers(const Token* tok)
     return tok;
 }
 
+const Token* skipTypeOrFunctionIdentifier(const Token* tok)
+{
+    if (tok == nullptr)
+        return nullptr;
+    while (tok->next() != nullptr && (tok->next()->str() == "::" || tok->next()->str() == "<"))
+    {
+        if (tok->next()->str() == "::")
+            tok = tok->tokAt(2);
+        else
+            tok = tok->linkAt(1);
+        if (tok == nullptr)
+            return nullptr;
+    }
+    return tok->next();
+}
+
+//Could this token be the first token of a call to a function instead of the declaration of a variable
+bool isFunctionCallInsteadOfVariableDeclaration(const Token* tok, const Scope& scope, const SymbolDatabase& db)
+{
+    auto start = tok;
+    tok = skipTypeOrFunctionIdentifier(tok);
+    if (tok == nullptr)
+        return false; //It's probably neither but it's not a function call
+    if (tok->str() != "(") //X y isn't a function call
+        return false;
+    if (tok->astOperand2() != nullptr && tok->astOperand2()->str() == ",")// X(a, b) isn't a variable declaration
```

Comment:
I _think_ this is purely an optimization and should always give the same result as findFunction though it does slightly improve things when the function declaration is not available (e.g. it's in a header cppcheck can't load).

Originally I asserted that if astOperand2 was null it must be a function because X() couldn't be a variable declaration. However this failed the unit test for ```extern const char ( * x [ 256 ] ) ;``` because the ast was flat. I suspect this is a bug in the ast but I'm not sure.